pkg/chunk/chunk_store.go

```diff
@@ -154,6 +160,14 @@ func (c *Store) Get(ctx context.Context, from, through model.Time, allMatchers .
 	if through < from {
 		return nil, fmt.Errorf("invalid query, through < from (%d < %d)", through, from)
 	}
+	if c.cfg.MaxQueryRangeDays == 0 {
+		// Specified as unlimited
+	} else {
```

Comment:
Yes shorter form is same logic ... but I regard longer form as more expressive to those who come after